game-headless/src/main/java/org/triplea/game/server/HeadlessGameRunner.java

```diff
@@ -17,6 +26,20 @@ public static void main(final String[] args) {
   }
 
   private static Injections constructInjections() {
-    return Injections.builder().engineVersion(new ProductVersionReader().getVersion()).build();
+    return Injections.builder()
+        .engineVersion(new ProductVersionReader().getVersion())
+        .playerTypes(gatherPlayerTypes())
+        .build();
+  }
+
+  private static Collection<PlayerTypes.Type> gatherPlayerTypes() {
+    return Stream.of(
+            PlayerTypes.getBuiltInPlayerTypes(),
+            List.of(new PlayerTypes.AiType(new DoesNothingAiProvider())),
+            StreamSupport.stream(ServiceLoader.load(AiProvider.class).spliterator(), false)
```

Comment:
I'm quite sure we're not going to deploy custom AIs to bots, for a number of reasons:
- none exist yet
- we keep bots uniform, if something is going to be baked into the deployment it is effectively part of the code
- we're fully responsible for bots running, any drop-in AIs are going to be us managing and debugging, at which point it really may as well be part of the main code base (a distributed code base is inherently more complex)
- AIs don't run on bots.

Reply:
I've removed the service provider code.